scripts/build_connectome/connectivity_plots.py

```diff
@@ -0,0 +1,854 @@
+                    f"connectivity measured as {task1_conn_measure}\n"
+                    f"and {task2} {task2_modality} connectivity measured\n"
+                    f"as {task2_conn_measure}"
+                )
+                if mean_center:
+                    fig_title = "Double mean centered " + fig_title
+                fig_xlabel = f"{task2} {task2_conn_measure}"
+                fig_ylabel = f"{task1} {task1_conn_measure}"
+
+                similarity_df = calculate_pearson_corr(
+                    [task1_conn, task2_conn],
+                    [list(task1_sub_ses.keys()), list(task2_sub_ses.keys())],
+                    mean_center,
+                )
+                plot_correlation_matrix(
+                    similarity_df,
+                    fig_title,
+                    fig_xlabel,
+                    fig_ylabel,
+                    name,
+                    mean_center,
+                )
+                d = save_all_info(
+                    similarity_df,
+                    d,
+                    task1,
+                    task2,
+                    task1_conn_measure,
+                    task2_conn_measure,
+                    task1_modality,
+                    task2_modality,
+                )
+                plt.close("all")
+
+    return d
+
+
+if __name__ == "__main__":
+    #### Inputs ####
+    # cache directory
+    cache = "/storage/store/work/haggarwa/"
+    tmp_dir = "connectivity_tmp"
+    # pairs of connectivity matrices to compare
+    # pick between DWI, RestingState and Raiders
+    task_pairs = [
+        ("RestingState", "DWI"),
+        ("Raiders", "DWI"),
+        ("DWI", "DWI"),
+        ("Raiders", "RestingState"),
+        ("Raiders", "Raiders"),
+        ("RestingState", "RestingState"),
+    ]
+    # calculated measures of functional connectivity
+    func_conn_measures = [
+        "Pearsons_corr",
+        "GraphicalLassoCV_corr",
+        "GraphicalLassoCV_partcorr",
+        "GroupSparseCovarianceCV_corr",
+        "GroupSparseCovarianceCV_partcorr",
+    ]
+    # calculated measures of structural connectivity
+    struc_conn_measures = [
+        "MNI152_siftweighted",
+        "individual_siftweighted",
+        "MNI152_nosift",
+        "individual_nosift",
+    ]
+    # get atlas
+    atlas = datasets.fetch_atlas_schaefer_2018(
+        data_dir=cache, resolution_mm=2, n_rois=400
+    )
+    # give atlas a custom name
+    atlas["name"] = "schaefer400"
+    # dictionary to store all the results
+    results = {
+        k: []
+        for k in [
+            "task1",
+            "task2",
+            "correlation",
+            "p_value",
+            "comparison",
+            "task1_measure",
+            "task2_measure",
+            "task1_modality",
+            "task2_modality",
+        ]
+    }
+
+    #### Run Comparisons ####
+    # save results with and without mean-centering the correlations
+    for mean_center in [True, False]:
```

Comment:
Same thing here, readability could e improved by modularizing.
In general, one should avoid functions longer than 50 lines.